### ISSUE TITLE
Property validUndefinedVariableNames is ignored inside else statement

### ISSUE THREAD

**Issue by udovicic** [sirbrillig/phpcs-variable-analysis#238]
Consider the following code to be part of some template, where variable `$block` originates from somewhere else. Its defined somewhere else, before `include` was called so it indeed is available in the template:

```php
<?php

$block->doSomething();

if (3  < 5) {
    $block->doThis();
} else {
    $block-doThat();
}
```

If you format ruelset to look like following:
```xml
    <rule ref="VariableAnalysis">
        <properties>
            <property name="validUndefinedVariableNames" value="block"/>
        </properties>
    </rule>
```

the output would be:
```
php,phtml test.phtml

FILE: /project/test.phtml
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 8 | WARNING | Variable $block is undefined.
----------------------------------------------------------------------

Time: 85ms; Memory: 8MB
```

Which is not the desired behavior.
Else statement requires special attention, but current implementation leads to false positives.

**Comment by sirbrillig** [comment]
Weird! I've reproduced in the test cases. Thanks for discovering this. I'll see what I can do.

In the meantime, you might find `allowUndefinedVariablesInFileScope` helpful for templates. It won't solve the problem if there are functions in there, and it means you won't get any warnings about undefined variables in the template, but it could be a compromise while this bug is being fixed.
